Missing util types

#633 added typings for `@embroider/util`, but this was later removed by #732. 
I wonder why? By accident maybe?

This now breaks upgrades for TS projects: `Cannot find module '@embroider/util' or its corresponding type declarations`.

--- a/packages/util/index.d.ts
+++ b/packages/util/index.d.ts
@@ -1,0 +1,4 @@
+export function ensureSafeComponent(
+  component: unknown,
+  thingWithOwner: unknown
+): unknown;

